### ISSUE TITLE
NumberInput - fix accessibility violations

### ISSUE THREAD

**Issue by thrbnhrtmnn** [deven-org/B01LER-Kitchen#774]
#### Description / User Story

The following errors and incomplete tests are thrown by the storybook accessibility addon and should be fixed:
- Violation: Buttons must have discernible text
- Violation: Form elements must have labels (placeholder="")
- Violation: Elements must meet minimum color contrast ratio thresholds (theme=dark)

----
#### Requirements
- [x] #534 

----
#### Acceptance Criteria
- [x] All errors and incomplete tests thrown by the accessibility addon are resolved:
   - [x] Violation: Buttons must have discernible text (readonly=false / unit=defined)
   - [x] Violation: Form elements must have labels (placeholder="")
   - [x] Violation: Elements must meet minimum color contrast ratio thresholds (theme=dark & unit=defined)

----
#### Background information
- ...

**Comment by thrbnhrtmnn** [comment]
Approved
